### PR TITLE
 Propagate uses_learning_phase via RNN initial states

### PR DESCRIPTION
**Pull request by myutwo150**
The states returned by an RNN layer with `return_state=True` do not carry the `uses_learning_phase` information. Therefore, for a seq2seq model whose decoder uses no dropout and receives only states from the encoder, `uses_learning_phase` is not set correctly.

To reproduce the error, the following lines
```python
encoder_input = Input((None, 5))
_, *states = LSTM(32, return_state=True, dropout=0.2)(encoder_input)
decoder_input = Input((None, 5))
decoder_output = LSTM(32)(decoder_input, initial_state=states)
model = Model([encoder_input, decoder_input], decoder_output)
model.predict([np.zeros((1, 2, 5)), np.zeros((1, 2, 5))])
```
will throw an error:
```
InvalidArgumentError (see above for traceback): You must feed a value for placeholder tensor 'lstm_1/keras_learning_phase' with dtype bool
	 [[Node: lstm_1/keras_learning_phase = Placeholder[dtype=DT_BOOL, shape=<unknown>, _device="/job:localhost/replica:0/task:0/device:GPU:0"]()]]
	 [[Node: lstm_2/TensorArrayReadV3/_19 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device_incarnation=1, tensor_name="edge_508_lstm_2/TensorArrayReadV3", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:CPU:0"]()]]
```
Fix it by adding `_uses_learning_phase` to the states returned by an RNN layer, and check if previous `state[0]` uses learning phase in the step functions.

### CODE REVIEW

File: keras/layers/recurrent.py [hunk 1]
```diff
@@ -893,7 +895,8 @@ def call(self, inputs, states, training=None):
             output = self.activation(output)
 
         # Properly set learning phase on output tensor.
-        if 0 < self.dropout + self.recurrent_dropout:
+        if (0 < self.dropout + self.recurrent_dropout or
+                getattr(states[0], '_uses_learning_phase', False)):
```
Comment (fchollet) [inline keras/layers/recurrent.py]: Why do you need to check the `_uses_learning_phase` attribute of the first state?

Comment (myutwo150) [reply keras/layers/recurrent.py]: You're right, no need to check this condition. I must have been tested on the wrong code. Sorry about that.
